### ISSUE TITLE
component testing suport

### ISSUE THREAD

**Issue by binvb** [cypress-io/code-coverage#596]
**Is your feature request related to a problem? Please describe.**
i can't find any document about code coverage width component testing, can code-coverage suport it?  thank you

**Comment by parky128** [comment]
+1 on this, anyone able to comment around supporting coverage for component tests?

**Comment by SystemParadox** [comment]
It seems to work if you follow the `[e2e coverage](https://docs.cypress.io/guides/tooling/code-coverage)` guide and just substitute `e2e` for `component` everywhere.

**Comment by ShellyDCMS** [comment]
@SystemParadox , Did you manage to collect coverage for Angular Cypress component tests? If so, could you share your repo?

**Comment by SystemParadox** [comment]
We use svelte not angular, but once you've got component tests working, following the e2e coverage guide should work, just substitute `e2e` for `component` everywhere.

**Comment by bindea-mihai** [comment]
@ShellyDCMS It's been a pain but managed to configure code coverage for cypress component tests. 
You can find a working solution [here](https://github.com/bindea-mihai/cypress-gherkin-ct/tree/code-coverage-tryout)

**Comment by jerkovicl** [comment]
@bindea-mihai  thx alot for this, it works with lib also

**Comment by ShellyDCMS** [comment]
@bindea-mihai , you rock!!! Thanks, man.

**Comment by jerkovicl** [comment]
@bindea-mihai @ShellyDCMS  Now we only need coverage for cypress e2e tests and we are golden

**Comment by bindea-mihai** [comment]
@jerkovicl [this](https://lukas-klement.medium.com/implementing-code-coverage-with-angular-and-cypress-6ed08ed7e617) is a very good tutorial to add coverage to e2e tests

**Comment by jerkovicl** [comment]
@bindea-mihai yeah i found that aswell and didnt work for me, Istanbul loader plugin didn't instrument code sadly

**Comment by ShellyDCMS** [comment]
@jerkovicl , the e2e tutorial didn't work for me as well at first, but then I took a look at the[ example repo](https://github.com/LukasKlement/angular-cypress-coverage-tutorial) (which has minor differences) and it worked.

**Comment by jerkovicl** [comment]
@ShellyDCMS thx for the tip, managed to set it up in the end

**Comment by TonySuSeismic** [comment]
I know that cypress plugin will use .nyc_output/out.json file to generate code coverage report, I tried below code for component testing but the file .nyc_output/out.json will not be updated.

`component: {
        setupNodeEvents(on, config) {
            require('@cypress/code-coverage/task')(on, config)
            return config
        },
        devServer: {
            framework: 'react',
            bundler: 'webpack',
            webpackConfig: require(webpackConfigFilePath),
        },
        specPattern: '**/*.cy.[jt]s(x)?',
        supportFile: 'cypress/support/component.tsx'
    }
`
    
For e2e it works what means after I finished e2e testing the file .nyc_output/out.json will be updated and code coverage report can be generated correctly.

`e2e: {
        setupNodeEvents(on, config) {
            require('@cypress/code-coverage/task')(on, config)
            return config
        },
        supportFile: 'cypress/support/e2e.ts'
    }
`

**Does anyone know its reason? It seems cypress/code-coverage/task just can support e2e testing, thanks**

**Comment by Benoit-Vasseur** [comment]
Hi,

I have the coverage "partialy working". I use Vite and and this plugin to instrument my code : https://www.npmjs.com/package/vite-plugin-istanbul
The branches are good but the numbers seems wrong. They are too high.

I have a single test with one mount but my report tells me that it runs twice :/.

Anyone with the same issue ?

**Comment by mwitty** [comment]
> I'm using Vite and have got the code instrumenting happening in the component tests (`window.__coverage__` exists) but there is no measurement/reporting of the coverage when running the component tests.
> 
> Edit: ah, the reason it wasn't working was because I missed the need for
> 
> ```js
> // cypress/component/support/component.js
> import '@cypress/code-coverage/support'
> ```

Thank you! I was also missing this.

**Comment by thevladisss** [comment]
Did anyone experience problem instrumenting code for component testing based on webpack? 

It works for E2E testing and code is properly instrumented as __coverage__ is available on window, while for component testing it is not.


I am using @quasar/quasar-app-extension-testing-e2e-cypress/cct-dev-server for my dev server which basically mimics webpack config that is used for build and development where __coverage__ available.

Any thoughts appreciated

**Comment by shrinidhimayya** [comment]
Not able to generate proper coverage info for cypress component tests. It says **"Could not find any coverage information in your application by looking at the window coverage object. Did you forget to instrument your application?"**
I am trying with svelte framework and bundler webpack. Documentation is all about e2e tests not finding info about component tests. Can someone post example of how do we instrument component tests with latest version of cypress : 13.10.0.

Thanks in advance.

**Comment by shrinidhimayya** [comment]
@ShellyDCMS , Thanks for sharing example. I am still getting error when I run component tests : "Could not find any coverage information in your application by looking at the window coverage object. Did you forget to instrument your application?" 

These are my set up : 
cypress : 13.10.0 , webpack : 5.91.0 , webpack-plugin-istanbul : 1.0.3 , @cypress/code-coverage: 3.12.39

**package.json:**

  "nyc": {
    "all": true,
    "include": "src/**/*.svelte"
  }

**cypress.config.js**

module.exports = defineConfig({
  component: {
    devServer: {
      framework: "svelte",
      bundler: "webpack",
    },
    specPattern: "src/**/*.test.{js,jsx,ts,tsx}",
    setupNodeEvents(on, config) {
      console.log("setupNodeEvents for components");
        require("@cypress/code-coverage/task")(on, config);
        return config;      
    },
  },
});

**webpack.config.js**

const { WebpackPluginIstanbul } = require("webpack-plugin-istanbul");

module.exports = {
  resolve: {
    // see below for an explanation
    mainFields: ["svelte", "browser", "module", "main"],
    conditionNames: ["svelte", "browser", "import"],
  },
  module: {
    rules: [
      {
        test: /\.(svelte)$/,
        exclude: /node_modules/,
        use: "svelte-loader",
      },
    ],
  },
  plugins: [
    new WebpackPluginIstanbul({
      include: ["src/**/*.svelte"],
      extension: [".svelte"],
      cwd: process.cwd()
    })
  ],
  devtool: "cheap-module-eval-source-map",
};

**cypress/support/component.js**

import '@cypress/code-coverage/support';

I think this file cypress/plugins/index.js is not required with latest version of cypress.

below is the screenshot : Also not able to see window.__coverage__ object

**Comment by ShellyDCMS** [comment]
@shrinidhimayya 
could you please try with plugin file, just to see if it makes a difference?

**Comment by shrinidhimayya** [comment]
> @shrinidhimayya could you please try with plugin file, just to see if it makes a difference?

it is expecting to have code coverage tasks inside cypress.config.js inside setupNodeEvents function. when this entry is added to only to cypress/plugins/index.js not inside setupnodeevents function then getting below error : "Code coverage tasks were not registered by the plugins file."
module.exports = (on, config) => {
  require("@cypress/code-coverage/task")(on, config);
  on(
    "file:preprocessor",
    require("@cypress/code-coverage/use-browserify-istanbul")
  );
  return config;
};

**Comment by ShellyDCMS** [comment]
@shrinidhimayya 
Please note that in the working example link I sent, it is done in the [plugin.ts](https://github.com/ShellyDCMS/cypress-test-utils-examples/blob/main/examples/lit/pokemon-catalog/cypress/plugins/index.ts) AND in the [cypress.config](https://github.com/ShellyDCMS/cypress-test-utils-examples/blob/c0f0869117f78bfb5f12a7fa72f91a6cf2b97850/examples/lit/pokemon-catalog/cypress.config.ts#L22) file.
Could you try the same?
